tiered-storage/jcloud/src/main/java/org/apache/bookkeeper/mledger/offload/jcloud/provider/TieredStorageConfiguration.java

```diff
@@ -0,0 +1,212 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.bookkeeper.mledger.offload.jcloud.provider;
+
+import com.google.common.collect.ImmutableMap;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+import org.jclouds.Constants;
+import org.jclouds.blobstore.BlobStore;
+import org.jclouds.domain.Credentials;
+import org.jclouds.providers.ProviderMetadata;
+
+/**
+ * Class responsible for holding all of the tiered storage configuration data
+ * that is set in the global Pulsar broker.conf file.
+ * <p>
+ * This class is used by the BlobStoreManagedLedgerOffloader to determine which
+ * JCloud provider to use for Tiered Storage offloand, along with the associated
+ * properties such as region, bucket, user credentials, etc.
+ * </p>
+ */
+public class TieredStorageConfiguration implements Serializable, Cloneable {
+
+    private static final long serialVersionUID = 1L;
+    public static final String BLOB_STORE_PROVIDER_KEY = "managedLedgerOffloadDriver";
+    public static final String METADATA_FIELD_BLOB_STORE_PROVIDER = "provider";
+    public static final String METADATA_FIELD_BUCKET = "bucket";
+    public static final String METADATA_FIELD_REGION = "region";
+    public static final String METADATA_FIELD_ENDPOINT = "endpoint";
+    protected static final int MB = 1024 * 1024;
+
+    public static TieredStorageConfiguration create(Properties props) throws IOException {
+        Map<String, String> map = new HashMap<String, String>();
+        map.putAll(props.entrySet()
```

Comment:
java.util.Properties already implements java.util.Map so the stream() stuff is unnecessary, and you can just do.
```
new TieredStorageConfiguration(Collections.unmodifiableMap(props));
```

Reply:
I tried your suggestion, but Collections.unmodifiableMap(props) returns a Map<Object,Object> which cannot be cast to Map<String,String>. Therefore it won't compile.